src/modules/project/sagas.ts

```diff
@@ -44,3 +53,19 @@ function* handleCreateProjectFromTemplate(action: CreateProjectFromTemplateActio
     onSuccess(project, scene)
   }
 }
+
+function* handleDuplicateProject(action: DuplicateProjectAction) {
+  const projects: Record<string, Project> = yield select(getProjects)
+  const originalProject = projects[action.payload.id]
+  if (!originalProject) return
+
+  const scenes: Record<string, SceneDefinition> = yield select(getScenes)
```

Comment:
Maybe it's better to use the return type of getScenes as type here and in `projects` above? Jus because if the type of the selector changes, we'll have a (difficult to trace) bug here

Reply:
✅ done